### ISSUE TITLE
Inconsistent GOSSIP_MAX_SIZE usage in GossipSub

### ISSUE THREAD

**Issue by Nashatyrev** [Consensys-Incorporated/teku#8957]
### Description

Teku is computing `compressedMaxGossipSize` based on uncompressed  `GOSSIP_MAX_SIZE` and [then passes it](https://github.com/Consensys/teku/blob/414e9091c8911757bb13acdd4c6c8d464a48f913/networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/config/LibP2PParamsFactory.java#L74) to Libp2p Gossip builder as a `maxGossipMessageSize` [parameter](https://github.com/libp2p/jvm-libp2p/blob/a00728e3ab469c46f3c915e8f5783aa92377f29e/libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipParams.kt#L81-L85). 

However `GossipRouter` [assumes](https://github.com/libp2p/jvm-libp2p/blob/e9e1d3392db113dfdb182d4bc51e1f0b6003ef77/libp2p/src/main/kotlin/io/libp2p/pubsub/AbstractRouter.kt#L115) this size as the overall gRPC message size, i.e. including all publish messages and control GossipSub messages.

This doesn't look like a security issue as there are no means to construct a valid gossip message of the `compressedMaxGossipSize`. Thus there is no ways to induce any kind of temporary network splits which may potentially happen when one client implementation refuses a valid gossip message while others propagate it.

This [spec PR](https://github.com/ethereum/consensus-specs/pull/4045) clarifies message size limits. The PR implementation should resolve this issue, so it probably makes sense to leave this issue until the PR is merged. 

### Versions
* Software version: current master version

**Comment by rolfyone** [comment]
So looking at this again now that the PR is done. Is your idea that we just pass the MaxGossipSize and libp2p handles compressed size @Nashatyrev ?

**Comment by Nashatyrev** [comment]
> So looking at this again now that the PR is done. Is your idea that we just pass the MaxGossipSize and libp2p handles compressed size [@Nashatyrev](https://github.com/Nashatyrev) ?

Not sure about this statement but the PR logic looks correct to me

**Comment by rolfyone** [comment]
yep all good, i figured it out when i was reviewing everything. ta.
